sketchlib/VectorSpace.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * A set of static functions that perform vector addition/scalar multiplication
+ * for some arbitrary type. For my purposes, the field is assumed to be number
+ * @template V The vector type
+ * @typedef {Object} VectorSpace
+ * @property {function(V, V): V} add
+ * @property {function(number, V): V} scale
+ */
+
+// ugh this will be a hard class to type...
+export class DirectSum {
```

Comment:
Let's not do this, the types are gross.

In practice, the simulations are all float-valued, so I can just use GeneralizedCoordinates